lib/auth/twitter_authenticator.rb

```diff
@@ -5,6 +5,10 @@ def name
     "twitter"
   end
 
+  def disable
+    SiteSetting.set_and_log(:enable_twitter_logins, false)
```

Comment:
My main issue with this `set_and_log` method is that there is no way to give it a reason. Could be nice to add an `opts` to this call, and then pass it down to `StaffActionLogger.log_site_setting_change`, to set a `context` value of something like "Automatically disabled after multiple failed health checks because of the `disable_failing_social_logins` site setting."

Also double checking, but if this failure happens for the 3 blips we show an admin problem on the dashboard right?

Reply:
Does the problem in the admin dashboard persist after Twitter auth is disabled? I think it needs to - otherwise admins might totally miss the fact we auto-disabled it? (it must be very common for admins not to visit a site for 3+ days)

Reply:
Good question; Ted I think this needs to happen too -- however I am not sure how we can swing this in our current problem checker system 🤔 Since we say `no_problem` if the authenticator is already disabled. Maybe we still do the healthy? check regardless? Not sure, something to think about.

Because it would be nice to have a problem saying "Twitter authentication is not working" showing on the dashboard even after it's been disabled automatically

Reply:
I think this will be straightforward once we add the `AdminMessage` model. PR coming soon.

Reply:
Sounds good,  I was wondering if we had to wait for that 👍